test/cmd-collect-source-warning.test.js

```diff
@@ -0,0 +1,18 @@
+'use strict'
+
+const test = require('tap').test
+const { spawn } = require('child_process')
+const endpoint = require('endpoint')
+const path = require('path')
+
+test('cmd - collect - source warning', function (t) {
+  const proc = spawn(process.execPath, [
+    path.join(__dirname, '/fixtures/sourcemap.run.js')
+  ])
```

Comment:
A check for the exit code can ensure that this won't accidentally pass if the subprocess fails:

```js
proc.on('exit', (code) => {
  t.strictEqual(code, 0)
})
```

The syntax error was obscured earlier because we didn't handle the case where the subprocess errored out. With this check, we'll know when the subprocess crashed, and can debug from there (for ex by running `node test/fixtures/sourcemap.run.js` which would show the full error message)